core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -758,9 +761,33 @@ private[spark] class BlockManager(
       blockId: BlockId,
       bytes: ChunkedByteBuffer,
       level: StorageLevel,
-      tellMaster: Boolean = true): Boolean = {
+      tellMaster: Boolean = true,
+      encrypt: Boolean = false): Boolean = {
```

Comment:
I think its worth documenting this param.  At first I was going to suggest that it should be called `allowEncryption` like the other one, but I realize its more complicated than that.  Maybe something like

If true, the given bytes should be encrypted before they are stored.  Note that in most cases, the given bytes will *already* be encrypted if encryption is on.  An important exception to this is with the streaming WAL.  Since the WAL does not support encryption, those bytes are generated un-encrypted.  But we still encrypt those bytes before storing in the block manager.

Maybe too wordy but I think its worth documenting.